### ISSUE TITLE
Allow a field to map to multiple different external field names when deserializing.

### ISSUE THREAD

**Issue by Lymia** [serde-rs/serde#1456]
I am parsing a file format where a field can appear with multiple different names when deserializing. Currently, it seems the only similar option for this is `#[serde(rename = "foo")]`, which no longer allows the field to be deserialized under its original name.

It would be nice to add a `#[serde(alias = "foo")]` attribute that would allow a field to be deserialized under multiple different names. For example:

```rust
#[derive(Deserialize)]
struct Struct {
    #[serde(alias = "foo")] bar: u32,
}
```

This feature would allow, for example, both `{"foo": 0}` and `{"bar": 0}` to deserialize to `Struct { bar: 0 }` when using `serde_json`. This seems like it would be relatively simple to implement, and I would be willing to PR an implementation of this if there's no problems with the feature idea.

**Comment by dtolnay** [comment]
How would you specify a field that is deserializable under two different names neither of which is its Rust field name?

Check out #595 which is maybe more intuitive in that case. I would accept a PR to support #595.

**Comment by Lymia** [comment]
My thought process was that it would be something like `#[serde(alias = "alias1", alias = "alias2", rename = "alias3")]`, but I the multiple rename syntax is certainly more intuitive when you don't want your Rust field name as a deserialize name.

I suppose my only objection to #595 is that it's somewhat clunky to use for my use case, because I'd have to write `#[serde(rename(deserialize = "foo", deserialize = "bar"))] foo: u32` for all the fields with aliases.

This isn't a huge deal in my case, because I only have a handful of them, but I can see it getting out of hand in formats with more. It could be an option to implement both, I guess, since many of the same code changes would support both.

**Comment by dtolnay** [comment]
Sounds good. I would accept an implementation where `serde(alias = "...")` is equivalent to `serde(rename(deserialize = "..."))` except that we keep accepting the original field name as long as all of the renames are specified by `alias` and not by `rename`.

**Comment by dtolnay** [comment]
Implemented in #1458. Thanks again!
